### PR TITLE
Split the project into lib and cli

### PR DESCRIPTION
**Pull request by Schaeff**
There is too much coupling between the CLI and the lib.
Start decoupling by having two cargo packages.

- [x] Introduce shell scripts for ease of set up with and without libsnark.
- [x] Extract the import resolver to remove fs access from zokrates_core
- [ ] decide on naming: should the CLI binary be called `zokrates-cli` or `zokrates` ?
- [ ] adjustments to README

### CODE REVIEW

**Comment by Schaeff** [comment]
@Kyroy any thoughts on how to deal with versioning if we split into a lib and a CLI?

**Comment by Kyroy** [comment]
There are 2 `.toml` files with 2 different versions. So there should be no problem.
I think we only dockerize the CLI anyway, right?

**Comment by Schaeff** [comment]
Oh right, if the version is only used to push to dockerhub we're fine yes

**Comment by Kyroy** [comment]
I just see that the version is `0.1.0`. In master we already have `0.2.0`. I think this commit never landed on develop :D

Also, you need to update https://github.com/Schaeff/ZoKrates/blob/d4b3873fa94022fa6bcda8f7dbd5701528d5a6ef/Jenkinsfile#L16 to get the version from the right `.toml` ;)

**Comment by Schaeff** [comment]
Hey @Kyroy this branch would have impact on CI, could you have a look? You should just be able to run `./test.sh` from now on

File: Dockerfile [hunk 1]
```diff
@@ -27,8 +30,12 @@ RUN curl https://sh.rustup.rs -sSf | \
 
 ENV PATH=/home/zokrates/.cargo/bin:$PATH
 
+WORKDIR /home/zokrates
```
[the remark this answers is not in the document]
Comment (Schaeff) [reply Dockerfile]: I'm very open to changes here, do you mean having a single /home/zokrates `WORKDIR` at the top? Would you want to remove the `$LIBSNARK_SOURCE_PATH` as well?

[the remark this answers is not in the document]
Comment (Kyroy) [reply Dockerfile]: No, I just wonder why you add `WORKDIR /root/` and `WORKDIR /home/zokrates` without any changes in the other lines

[the remark this answers is not in the document]
Comment (kyroy) [reply Dockerfile]: Same holds still for this line. It is not required ;)

**Pull request merged**